lib/elixir/lib/calendar.ex

```diff
@@ -293,6 +293,29 @@ defmodule Date do
     end
   end
 
+  @doc """
+  Compares two `Date` structs.
+
+  Returns :gt if first date is later than the second
+  and :lt for vice versa. If the two dates are equal
+  :eq is returned
+
+  ## Examples
+
+      iex> Date.compare(Date.from_erl!({2016, 4, 16}), Date.from_erl!({2016, 4, 28}))
+      :lt
```

Comment:
WDYT about using sigils in this and other examples?

```elixir
iex> Date.compare(~D[2016-04-16], ~D[2016-04-28])
:lt
```

If sounds good, I'd be happy to open up a PR for that change